src/rpc/blockchain.cpp

```diff
@@ -285,7 +285,7 @@ UniValue blockToJSON(const CBlock& block, const CBlockIndex* blockindex, bool tx
 
 UniValue getblockcount(const UniValue& params, bool fHelp)
 {
-    if (fHelp || params.size() != 0)
+    if (fHelp)
```

Comment:
Ah, I misunderstood this comment until I got to the end. You’re right that `zcash-cli` will now just error about not having the correct number of arguments, but if a request _is_ made to the server with the wrong arg count (say, the Python API or some other client), they’ll get the help message as before (since we just set `fHelp` to `true` if there is an incorrect param length).

However, one of the reasons this is still in draft is that there are cases where we use `fHelp` for other purposes, and I want to make sure that those slight changes in semantics are ok. But I didn’t want to put more work into this until after syncing with @str4d on how this fits into the future of the RPC API (e.g., supporting keyed arguments).